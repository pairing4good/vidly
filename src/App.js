import React from "react";
import "./App.css";
import Movies from "./components/movies";
import "@fortawesome/fontawesome-svg-core/styles.css";

function App() {
  return (
    <main className="container">
      <Movies />
    </main>
  );
}

export default App;
