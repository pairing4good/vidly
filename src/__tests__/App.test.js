import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import App from "../App";

test("should match the intial page render snapshot", () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});

test("should match after movie deletion", () => {
  const { container, getByText } = render(<App />);
  const deleteButtons = screen.getAllByText("Delete");

  fireEvent.click(deleteButtons[0]);

  expect(getByText("Showing 8 movies in the database.")).toBeTruthy();

  expect(container.firstChild).toMatchSnapshot();
});

test("should match after all movies deleted", () => {
  const { container, getAllByText, getByText } = render(<App />);
  let deleteButtons = getAllByText(`Delete`);

  expect(getByText("Showing 9 movies in the database.")).toBeTruthy();

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 8 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 7 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 6 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 5 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 4 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 3 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 2 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("Showing 1 movies in the database.")).toBeTruthy();
  deleteButtons = getAllByText(`Delete`);

  fireEvent.click(deleteButtons[0]);
  expect(getByText("There are no movies in the database.")).toBeTruthy();

  expect(container.firstChild).toMatchSnapshot();
});
