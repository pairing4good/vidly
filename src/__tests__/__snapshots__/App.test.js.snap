// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match after all movies deleted 1`] = `
<main
  class="container"
>
  <div>
    There are no movies in the database.
  </div>
</main>
`;

exports[`should match after movie deletion 1`] = `
<main
  class="container"
>
  <div>
    Showing 
    8
     movies in the database.
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Genere
        </th>
        <th
          scope="col"
        >
          Stock
        </th>
        <th
          scope="col"
        >
          Rate
        </th>
        <th
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Die Hard
        </td>
        <td>
          Action
        </td>
        <td>
          5
        </td>
        <td>
          2.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Get Out
        </td>
        <td>
          Thriller
        </td>
        <td>
          8
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Trip to Italy
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Airplane
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Wedding Crashers
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Gone Girl
        </td>
        <td>
          Thriller
        </td>
        <td>
          7
        </td>
        <td>
          4.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          The Sixth Sense
        </td>
        <td>
          Thriller
        </td>
        <td>
          4
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          The Avengers
        </td>
        <td>
          Action
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</main>
`;

exports[`should match the intial page render snapshot 1`] = `
<main
  class="container"
>
  <div>
    Showing 
    9
     movies in the database.
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Genere
        </th>
        <th
          scope="col"
        >
          Stock
        </th>
        <th
          scope="col"
        >
          Rate
        </th>
        <th
          scope="col"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Terminator
        </td>
        <td>
          Action
        </td>
        <td>
          6
        </td>
        <td>
          2.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Die Hard
        </td>
        <td>
          Action
        </td>
        <td>
          5
        </td>
        <td>
          2.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Get Out
        </td>
        <td>
          Thriller
        </td>
        <td>
          8
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Trip to Italy
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Airplane
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Wedding Crashers
        </td>
        <td>
          Comedy
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Gone Girl
        </td>
        <td>
          Thriller
        </td>
        <td>
          7
        </td>
        <td>
          4.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          The Sixth Sense
        </td>
        <td>
          Thriller
        </td>
        <td>
          4
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          The Avengers
        </td>
        <td>
          Action
        </td>
        <td>
          7
        </td>
        <td>
          3.5
        </td>
        <td>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</main>
`;
